### ISSUE TITLE
MockFile.ReadAllBytes throws NullReferenceException

### ISSUE THREAD

**Issue by dgee2** [System-IO-Abstractions/System.IO.Abstractions#173]
Hi When testing some logic which is dealing with a situation where we are trying to read a file that does not exist, I am receiving a NullReferenceException instead of a FileNotFoundException. This appears to be a bug with the library.

ReadAllBytes appears to be missing the FileExists check which is included in the other read methods. I suggest adding the following code (sorry I can't submit a pull request atm)

```
if (!mockFileDataAccessor.FileExists(path))
{
    throw new FileNotFoundException(string.Format(CultureInfo.InvariantCulture, "Can't find {0}", path));
}
```

**Comment by manne** [comment]
yes, it is a bug

**Comment by dgee2** [comment]
I've added a pull request for this. Would appreciate this being pushed through to Nuget and at some point hopefully I'll find more time to have an in depth look at some of the mocking classes to make sure they're matching the System.IO implementation (as far as exceptions are concerned)

**Comment by jpreese** [comment]
Will be resolved by #301

**Comment by jpreese** [comment]
This has been resolved in the latest release.
